### PR TITLE
fix: update aws.exports file on amplify push

### PR DESCRIPTION
**Pull request by akshbhu**
*Issue #6292 *

*Description of changes:*
* added localMeta obj to generate current AWSexports
* Removed logic to copy cloudMeta and oldAWSexports file
* added tests for the same

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-frontend-javascript/lib/frontend-config-creator.js [hunk 1]
```diff
@@ -339,7 +329,8 @@ function getAPIGWConfig(apigwResources, projectRegion, configOutput) {
   };
 
   for (let i = 0; i < apigwResources.length; i += 1) {
-    if (apigwResources[i].output.ApiName && apigwResources[i].output.RootUrl) { // only REST endpoints contains this information
+    if (apigwResources[i].output.ApiName && apigwResources[i].output.RootUrl) {
```
Comment (jhockett) [inline packages/amplify-frontend-javascript/lib/frontend-config-creator.js:332]: nit: this for loop can be simplified (I didn't test this snippet)

```ts
apigwConfig.aws_ckoud_logic_custom = apigwResources
  .filter(r => r.output.ApiName && r.output.RootUrl)
  .map(r => {
    return {
        name: r.output.ApiName,
        endpoint: r.output.RootUrl,
        region: projectRegion,
    }
  });
return apigwConfig;
```

File: packages/amplify-frontend-javascript/lib/frontend-config-creator.js [hunk 2]
```diff
@@ -75,28 +75,18 @@ function createAmplifyConfig(context, amplifyResources) {
...
-  if (currentAWSExports) {
-    Object.keys(currentAWSExports)
-      .filter(key => !CUSTOM_CONFIG_BLACK_LIST.includes(key))
-      .forEach(key => {
-        if (!cloudAWSExports[key]) {
-          customConfigs[key] = currentAWSExports[key];
-        }
-      });
+  if (newAWSExports) {
+    Object.keys(newAWSExports).filter(key => !CUSTOM_CONFIG_BLACK_LIST.includes(key));
```
Comment (jhockett) [inline packages/amplify-frontend-javascript/lib/frontend-config-creator.js:89]: Isn't this now essentially a noop? Object.keys returns an array, but no variable is assigned here

File: packages/amplify-frontend-javascript/lib/frontend-config-creator.js (hunk 1 above)
Comment (akshbhu) [reply packages/amplify-frontend-javascript/lib/frontend-config-creator.js]: Thanks!! , works fine , tested it on my end

File: packages/amplify-frontend-javascript/lib/frontend-config-creator.js (hunk 2 above)
Comment (akshbhu) [reply packages/amplify-frontend-javascript/lib/frontend-config-creator.js:87]: resolved

Comment (jhockett) [reply packages/amplify-frontend-javascript/lib/frontend-config-creator.js:87]: Sorry, I was referring to this line: `Object.keys(newAWSExports).filter(key => !CUSTOM_CONFIG_BLACK_LIST.includes(key));`

It doesn't look like it does anything

Comment (edwardfoyle) [reply packages/amplify-frontend-javascript/lib/frontend-config-creator.js:87]: also, rename `CUSTOM_CONFIG_BLACK_LIST` to `CUSTOM_CONFIG_BLOCK_LIST` or `CUSTOM_CONFIG_DENY_LIST`

**Comment by rajrajhans** [comment]
This PR closes #6432 as well.

**Comment by renebrandel** [comment]
@kaustavghosh06  - can you review thi?

**Comment by josefaidt** [comment]
Per @Straubulous 's comment above we'll be closing this due to age (>1 year) and inactivity, and will be revisited in the future 🙂

**Pull request closed without merging**